common/utils/src/main/java/org/graalvm/buildtools/agent/AgentConfiguration.java

```diff
@@ -143,41 +144,29 @@ private void addDefaultAccessFilter() {
             // this could only happen if we instantiated disabled agent configuration
             return;
         }
+        Path defaultAgentConfigTempFile = createDefaultAgentConfigTempFile();
+        try {
+            Files.writeString(defaultAgentConfigTempFile, getDefaultAgentConfigContent());
+            accessFilterFiles.add(defaultAgentConfigTempFile.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot write default access filter to temporary file "+ defaultAgentConfigTempFile.toAbsolutePath(), e);
+        }
+    }
 
-        String tempDir = System.getProperty("java.io.tmpdir");
-        Path agentDir = Path.of(tempDir).resolve("agent-config");
-        Path accessFilterFile = agentDir.resolve(ACCESS_FILTER_PREFIX + ACCESS_FILTER_SUFFIX);
-        if (Files.exists(accessFilterFile)) {
-            accessFilterFiles.add(accessFilterFile.toString());
-            return;
+    private static Path createDefaultAgentConfigTempFile() {
+        try {
+            return Files.createTempFile(ACCESS_FILTER_PREFIX, ACCESS_FILTER_SUFFIX);
```

Comment:
I do not see those temp files getting deleted afterwards. This will accumulate lots of files in the temp dir over time.

Reply:
I have updated the pull request to delete the file on exit.